transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,213 @@
+	return t.Config, t.Env
+}
+
+// findMainScript returns the path of .py file having the main function
+func findMainScript(pythonFilesPath []string) (string, error) {
+	if len(pythonFilesPath) == 0 {
+		return "", nil
+	}
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := os.Open(pythonFilePath)
+		if err != nil {
+			logrus.Debugf("Failed to open the file %s", pythonFilePath)
+		}
+		scanner := bufio.NewScanner(pythonFile)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			if pythonMainRegex.MatchString(scanner.Text()) {
+				return pythonFilePath, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("could not find the main function in python files")
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+	pythonFiles, err := common.GetFilesByExt(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilePath, err := findMainScript(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	// Search requirements.txt
+	if mainPythonFilePath != "" {
+		serviceName := strings.TrimSuffix(filepath.Base(mainPythonFilePath), pythonExt)
+		if serviceName == pythonMain {
+			serviceName = pythonService
+		} else if serviceName == pythonManage {
+			serviceName = djangoService
+		}
+		services = map[string][]transformertypes.Artifact{
+			serviceName: {{
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+					MainPythonFilePathType:        {mainPythonFilePath},
+				},
+			}},
+		}
+		return services, nil
+	}
+	return nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PythonDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var pythonConfig PythonTemplateConfig
+		pythonConfig.AppName = a.Name
+		if a.Name == djangoService {
```

Comment:
This should be passed as part of a config in artifact and should not be inferred from service name.

Reply:
Done.